compact.go

```diff
@@ -784,7 +784,7 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 				return errors.Wrap(err, "merge overlapping chunks")
 			}
 		}
-		if err := chunkw.WriteChunks(b[:], mergedChks...); err != nil {
+		if b, err = chunkw.WriteChunks(b[:0], mergedChks...); err != nil {
```

Comment:
This is tricky, easy to make mistake and add `:=` and start shadowing `b`. I might think we don't need to return b (see comments above) but if we do separate `if err != nil` might be better

Reply:
As `WriteChunks` is an interface, I would not make any assumptions of how much bytes does it allocate (though we can see it easily). So design-wise returning should be fine, which would either be the same slice or a bigger slice.

Reply:
Just checked that you cannot have `:=` here, as `b` would be declared and never used. Do you still prefer to be in a separate line?

Reply:
Done